api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -605,7 +524,7 @@ def __init__(
             if len(encoders) != len({enc.name for enc in encoders}):
                 raise ValueError("Encoders must have unique names")
 
-            self.obs_column_names = list(itertools.chain(*[enc.columns for enc in encoders]))
+            self.obs_column_names = list(set(itertools.chain(*[enc.columns for enc in encoders])))
```

Comment:
`set`s don't have a consistent iteration order, which can be a source of really hard to figure out bugs. I'd suggest deduplication in an order-preserving way.

Stack overflow says:

```suggestion
            self.obs_column_names = list(dict.fromkeys(itertools.chain(*[enc.columns for enc in encoders])))
```

https://stackoverflow.com/questions/39755464/python-unique-items-of-list-in-the-order-that-it-appears